src/sql/workbench/api/node/extHostModelViewDialog.ts

```diff
@@ -23,24 +23,66 @@ class DialogImpl implements sqlops.window.modelviewdialog.Dialog {
 	public customButtons: sqlops.window.modelviewdialog.Button[];
 	public readonly onValidityChanged: vscode.Event<boolean>;
 	private _valid: boolean = true;
+	private _modelView: sqlops.ModelView;
+	public handle: number;
 
-	constructor(private _extHostModelViewDialog: ExtHostModelViewDialog) {
+	constructor(private _extHostModelViewDialog: ExtHostModelViewDialog,
+		private _extHostModelView: ExtHostModelViewShape) {
 		this.okButton = this._extHostModelViewDialog.createButton(nls.localize('dialogOkLabel', 'Done'));
 		this.cancelButton = this._extHostModelViewDialog.createButton(nls.localize('dialogCancelLabel', 'Cancel'));
 		this.onValidityChanged = this._extHostModelViewDialog.getValidityChangedEvent(this);
 		this.onValidityChanged(valid => this._valid = valid);
 	}
 
+	public registerContent(handler: (view: sqlops.ModelView) => void): void {
+		let viewId = 'dialog' + this.handle;
+		this.content = viewId;
+			this._extHostModelView.$registerProvider(viewId, modelView => {
+				this._modelView = modelView;
+				handler(modelView);
+			});
+	}
+
 	public get valid(): boolean {
 		return this._valid;
 	}
+
+	public get modelView(): sqlops.ModelView {
+		return this._modelView;
+	}
+
+	public set modelView(value: sqlops.ModelView) {
+		this._modelView = value;
+	}
 }
 
 class TabImpl implements sqlops.window.modelviewdialog.DialogTab {
 	public title: string;
 	public content: string;
+	private _modelView: sqlops.ModelView;
+
+	public handle: number;
 
-	constructor(private _extHostModelViewDialog: ExtHostModelViewDialog) { }
+	public get modelView(): sqlops.ModelView {
+		return this._modelView;
+	}
+
+	public set modelView(value: sqlops.ModelView) {
+		this._modelView = value;
+	}
+
+	public registerContent(handler: (view: sqlops.ModelView) => void): void {
+		let viewId = 'tab' + this.handle;
+		this.content = viewId;
+			this._extHostModelView.$registerProvider(viewId, modelView => {
```

Comment:
Minor: indentation looks wrong here. Also another opportunity to put this code in a base class?